awx/sso/middleware.py

```diff
@@ -62,6 +58,16 @@ def process_request(self, request):
                 if auth_token and request.user and request.user.is_authenticated():
                     request.session.pop('social_auth_error', None)
                     request.session.pop('social_auth_last_backend', None)
+            else:
+                if request.user and request.user.is_authenticated():
+                    # The rest of the code base rely hevily on type/inheritance checks,
+                    # LazyObject sent from Django auth middleware can be buggy if not
+                    # converted back to its original object.
+                    if isinstance(request.user, LazyObject) and \
```

Comment:
I'm curious what the `LazyObject` breaks -- I don't recall ever having issues with `request.user` as a result of it.